### PR TITLE
DP-4174 document the release process

### PR DESCRIPTION
**Pull request by jesconstantine**
This PR adds documentation for the production release process.

JIRA ticket: https://jira.state.ma.us/browse/DP-4174

TO TEST:
- [ ] Read the proposed documentation and verify that they are clear.

Notes:

The technical process is heavily based on the developer deploy script, which has already been tested.  Developers and Release managers will test the workflow next sprint.

### CODE REVIEW

File: docs/release.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Release to Production
+Mayflower release managers with the necessary repo permissions can follow these steps to deploy code to production (i.e. do a release).
+
+*Note: this assumes that your local environment is already set up and functioning.  If this is not true, follow the steps in [Getting Started](../.github/CONTRIBUTING.md#getting-started).*
+
+1. If there is new code to be delivered, notify the team at least two hours ahead of time that a release is coming. Follow the [Communicate Releases](https://wiki.state.ma.us/display/massgovredesign/Communicating+Releases) instructions for Upcoming Deployments.
+1. Pull and checkout the [massgov/mayflower `dev` branch](https://github.com/massgov/mayflower/commits/dev): `git fetch upstream && git checkout dev`.
```
Comment (isaacchansky) [inline docs/release.md]: assumption is that you have a remote named 'upstream' here, idk if you mention this elsewhere in docs but might be good to point out here.

Comment (jesconstantine) [reply docs/release.md]: @isaacchansky Thanks - I'll throw a link to the docs where we walk people through setting up their local repositories with massgov/mayflower as `upstream` -- would that address your concern?

**Comment by powpow12** [comment]
@jesconstantine these are not mandatory, just my brain dump.

Just posting some one offs here, I viewed the release.docs in the finished state,

>Mayflower is the enterprise design system for the Commonwealth of Massachusetts.

Would / should I know this is pattern lab based?

> Follow the steps in Demo Install to get a copy of the project up and running on your local machine for demo and testing purposes.

I like it when we use fun/playful text, maybe saying something on the lines of:  _For example, if you want to show off our awesome design library to business partners, this is the route for you :)_  

Possible text change:

>Please see Deployment docs for steps on deploying development work to a Mayflower fork's Github Pages.

to 

> Please see Deployment docs for steps on deploying development work to a fork of Mayflower's Github Pages.

Regarding Mayflower assets, should we mention that other css flavor we use? What is it scss?

**Comment by powpow12** [comment]
Re Setting up your environment,

Is there a test one can do to confirm they have there environment set up right? If trying to install mayflower is the only "test" that's fine just didn't know if we could capture anything here that would break something down the road. Obviously, this could get rather verbose, I mean it in the MVP sense.

Re Demo Install

Maybe add a last step that say something like, we are hard at work adding new features and if you want to make sure you have the latest and greatest don't forget to pull often and rebuild.

Re Contributing to mayflower

> help wanted - Issues we need or would love help from the community to resolve.

If I was a contributor and I didn't have the time to read these detailed explanation of labels, I may think this is a general support label.  We could possibly update to *Request for community contributor* but even that sounds too machine or not exactly what we want.

Do we have a mayflower issue template? If not, that would be extremely helpful to "force" a format we want.

>Please do not edit the gh-pages branch. That branch is generated and used to deploy work for testing.

Can users without collaborator permission do this?

> Please do not edit the gh-pages branch. That branch is generated and used to deploy work for testing.

File: docs/release.md (hunk 1 above)
Comment (jesconstantine) [reply docs/release.md]: Clarified / called out in https://github.com/massgov/mayflower/pull/531/commits/f7e4963a0648b482774ce94997fe1d88446d7cb5

**Comment by jesconstantine** [comment]
Thanks so much for your time and suggestions here @powpow12 

Here are my take aways:
- clearer label for "help wanted" = "PRs welcome" https://github.com/massgov/mayflower/pull/531/commits/9065414eaa17edb502e4dcea3884886ea3663e15
- remind demo install'ers to `git pull` before they `gulp` https://github.com/massgov/mayflower/pull/531/commits/d53fe00ee25fb3190b39d33fb8b48e07cd9315f1
- make the line re: deploy docs easier to understand Mayflower fork's -> fork of Mayflower's https://github.com/massgov/mayflower/pull/531/commits/db703bece8be3f69a32ad2b7db70b44b1ec11b16

- re: referencing pattern lab in the first line of the repo readme, I don't think so.  we do call it out specifically in the ["built with" section on the readme](https://github.com/jesconstantine/mayflower/blob/c98bcd1bb7cf381f540830a9687c2ad5e999d9cc/README.md) -- thoughts?
- re: Mayflower assets and scss vs css - the assets refer to built / compiled assets so it will really be more about css there.  We do use scss which I've documented at a very high level (more to come) and just recently added a link to sass docs in https://github.com/massgov/mayflower/pull/531/commits/c98bcd1bb7cf381f540830a9687c2ad5e999d9cc

**Pull request merged**